toolkit/tools/pkg/isomakerlib/isomaker.go

```diff
@@ -276,16 +321,38 @@ func (im *IsoMaker) prepareWorkDirectory() {
 // copyStaticIsoRootFiles copies architecture-independent files from the
 // Mariner repo directories.
 func (im *IsoMaker) copyStaticIsoRootFiles() {
-	staticIsoRootFilesPath := filepath.Join(im.resourcesDirPath, "assets/isomaker/iso_root_static_files/*")
 
-	logger.Log.Debugf("Copying static ISO root files from '%s'.", staticIsoRootFilesPath)
+	if im.resourcesDirPath == "" && im.grubCfgPath == "" {
+		logger.Log.Panicf("Missing required parameters. Must specify either the resources directory or provide a grub.cfg.")
+	}
+
+	if im.resourcesDirPath != "" {
+		staticIsoRootFilesPath := filepath.Join(im.resourcesDirPath, "assets/isomaker/iso_root_static_files/*")
+
+		logger.Log.Infof("Copying static ISO root files from '%s' to '%s'.", staticIsoRootFilesPath, im.buildDirPath)
 
-	recursiveCopyDereferencingLinks(staticIsoRootFilesPath, im.buildDirPath)
+		recursiveCopyDereferencingLinks(staticIsoRootFilesPath, im.buildDirPath)
+	}
+
+	// im.grubCfgPath allows the user to overwrite the default grub.cfg that is
+	// copied from the resource folder.
+	if im.grubCfgPath != "" {
+		targetGrubCfg := filepath.Join(im.buildDirPath, "boot/grub2/grub.cfg")
+		targetGrubCfgDir := filepath.Dir(targetGrubCfg)
+		logger.PanicOnError(os.MkdirAll(targetGrubCfgDir, os.ModePerm), "Failed while creating directory '%s'.", targetGrubCfgDir)
+
+		logger.Log.Infof("Copying '%s' to '%s'.", im.grubCfgPath, targetGrubCfg)
+		shell.MustExecuteLive("cp", im.grubCfgPath, targetGrubCfg)
+	}
 }
 
 // copyArchitectureDependentIsoRootFiles copies the pre-built UEFI modules required
 // to boot the ISO image.
 func (im *IsoMaker) copyArchitectureDependentIsoRootFiles() {
+	if im.resourcesDirPath == "" || !im.enableBiosBoot {
```

Comment:
Could you add a comment explaining why we want to exit when this condition is true?

Reply:
updated.